File: app/src/main/java/com/chemecador/secretaria/requests/TaskRequest.java
package com.chemecador.secretaria.requests;

import com.chemecador.secretaria.items.Task;

import java.time.format.DateTimeFormatter;

/**
 * 05/08/2023
 * Clase TaskRequest
 *
 * */

public class TaskRequest {


    private String title;
    private String content;
    private String start_time;
    private int id;


    public TaskRequest(Task task) {
        this.id = task.getId();
        this.title = task.getTitle();
        this.content = task.getContent();

        // Formatear la fecha y hora en el formato deseado
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        this.start_time = task.getStartTime().format(formatter);
    }
}

File: app/src/main/java/com/chemecador/secretaria/requests/TaskRequest.kt
package com.chemecador.secretaria.requests

import com.chemecador.secretaria.items.Task
import com.google.gson.annotations.SerializedName
import java.time.format.DateTimeFormatter

/**
 * 05/08/2023
 * Clase TaskRequest
 *
 */
class TaskRequest(task: Task) {
    private val title: String?
    private val content: String?
    @field:SerializedName("start_time")
    private val startTime: String
    private val id: Int

    init {
        id = task.id
        title = task.title
        content = task.content

        // Formatear la fecha y hora en el formato deseado
        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
        startTime = task.startTime!!.format(formatter)
    }
}